File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/examples/loginview/de/vimotest/examples/loginview/LoginViewModel.cs
/// <filename>
///     LoginViewModel.cs
/// </filename>
namespace de.vimotest.examples.loginview
{
    public class LoginViewModel
    {
        private string UsernameTextBoxText;

        public virtual string getUsernameTextBoxText()
        {
            return this.UsernameTextBoxText;
        }

        private string PasswordTextBoxText;

        public virtual string getPasswordTextBoxText()
        {
            return this.PasswordTextBoxText;
        }

        private bool isShowPasswordCheckBoxChecked;

        public virtual bool getIsShowPasswordCheckBoxChecked()
        {
            return this.isShowPasswordCheckBoxChecked;
        }

        private bool isLoginButtonEnabled;

        public virtual bool getIsLoginButtonEnabled()
        {
            return this.isLoginButtonEnabled;
        }

        private string ErrorLabelText;

        public virtual string getErrorLabelText()
        {
            return this.ErrorLabelText;
        }

        private bool isErrorLabelVisible;

        public virtual bool getIsErrorLabelVisible()
        {
            return this.isErrorLabelVisible;
        }

        public virtual void loadViewModel()
        {
        }

        public virtual void usernameTextBoxTextFilled(string text)
        {
        }

        public virtual void passwordTextBoxTextFilled(string text)
        {
        }

        public virtual void showPasswordCheckBoxChecked(bool isChecked)
        {
        }

        public virtual void loginButtonClicked()
        {
        }
    }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/examples/loginview/de/vimotest/examples/loginview/LoginViewModel.cpp
#include "LoginViewModel.h"
#include <string>

namespace de::vimotest::examples::loginview
{
  std::string LoginViewModel::getUsernameTextBoxText()
  {
    return this->UsernameTextBoxText;
  }
  std::string LoginViewModel::getPasswordTextBoxText()
  {
    return this->PasswordTextBoxText;
  }
  bool LoginViewModel::getIsShowPasswordCheckBoxChecked()
  {
    return this->isShowPasswordCheckBoxChecked;
  }
  bool LoginViewModel::getIsLoginButtonEnabled()
  {
    return this->isLoginButtonEnabled;
  }
  std::string LoginViewModel::getErrorLabelText()
  {
    return this->ErrorLabelText;
  }
  bool LoginViewModel::getIsErrorLabelVisible()
  {
    return this->isErrorLabelVisible;
  }
  void LoginViewModel::loadViewModel()
  {
  }
  void LoginViewModel::usernameTextBoxTextFilled(std::string text)
  {
  }
  void LoginViewModel::passwordTextBoxTextFilled(std::string text)
  {
  }
  void LoginViewModel::showPasswordCheckBoxChecked(bool isChecked)
  {
  }
  void LoginViewModel::loginButtonClicked()
  {
  }
}
